presto-main/src/main/java/com/facebook/presto/operator/SetBuilderOperator.java

```diff
@@ -133,7 +134,9 @@ public OperatorFactory duplicate()
 
     private final ChannelSetBuilder channelSetBuilder;
 
-    private boolean finished;
+    private boolean finished;  // whether there are no more input pages
+
+    private Optional<Work<?>> unfinishedWork;  // The pending work for current page.
```

Comment:
I usually prefer avoid using `Optional` if the field is completely local. `Optional` is used when the variable is passed in or returned to inform the other component that the field could be null. But it's ok to use `Optional` if you prefer. If you decide to go with `Optional`, make it final; otherwise, annotate it with `@Nullable`.

Reply:
@highker it was my vote to make it `Optional` (and it can't be `final`). I know `Optional` is more cumbersome to use, but it remains easier not to forget about it being.. optional.

Reply:
@findepi, as mentioned above, I'm totally ok using `Optional`; it's just more like my personal choice using `Optional` only when as a part of an interface. Right, it can't be final.

Reply:
This remains a matter of preference, I know. I'm ok with nullable. I said that it was by vote, so that code author doesn't feel oppressed by two reviewers playing ping-pong ;)